### ISSUE TITLE
[HTML Producer] Audio not routed to consumers

### ISSUE THREAD

**Issue by Julusian** [CasparCG/Server#669]
CEF does not appear to support routing of audio, there is a PR to add it but it is stale and not near being merged: https://bitbucket.org/chromiumembedded/cef/pull-requests/18/adding-audio-support/diff
This means that any audio played in it gets sent to the system default soundcard, rather than any audio consumer for the channel. It also means that any audio mixer commands will have no effect.

Once CEF supports routing of audio we should update our html producer to make use of it.

**Comment by premultiply** [comment]
Yes, very important feature for HTML template producer.
Thanks for pointing on that.

**Comment by Julusian** [comment]
A new CEF PR has been raised, which replaces the previous one.
https://bitbucket.org/chromiumembedded/cef/pull-requests/151/add-ability-to-redirect-audio-output-to/diff

**Comment by premultiply** [comment]
Is it possible to merge that?

**Comment by ronag** [comment]
@premultiply we would need a custom CEF build for that.

**Comment by Julusian** [comment]
I'm keeping a close eye on it. Once I have some time I intend to test it out in Caspar, but I think its a bit too late to add that into 2.2

**Comment by Julusian** [comment]
https://bitbucket.org/chromiumembedded/cef/pull-requests/177/add-ability-to-capture-audio-output-to/diff

**Comment by Jgaleas7** [comment]
Any solution for this or any advance?

**Comment by Jgaleas7** [comment]
> https://bitbucket.org/chromiumembedded/cef/pull-requests/177/add-ability-to-capture-audio-output-to/diff

it was DECLINED  :(

**Comment by Julusian** [comment]
@Jgaleas7 no, it was merged:
> Marshall Greenblatt
Manually merged in master revision 58e1149.

https://bitbucket.org/chromiumembedded/cef/commits/58e1149c7127314072903d3d45b9ba8b9fd2fc92

It think that should be available in official CEF builds for Chromium 74. I'll keep an eye out for when builds become available that we can use.

**Comment by jesperstarkar** [comment]
Oooooooooooooohhhhh!!!!!!!

**Comment by Jgaleas7** [comment]
> @Jgaleas7 no, it was merged:
> 
> > Marshall Greenblatt
> > Manually merged in master revision 58e1149.
> 
> https://bitbucket.org/chromiumembedded/cef/commits/58e1149c7127314072903d3d45b9ba8b9fd2fc92
> 
> It think that should be available in official CEF builds for Chromium 74. I'll keep an eye out for when builds become available that we can use.

CEF chromium 74 is available http://opensource.spotify.com/cefbuilds/index.html

**Comment by Jgaleas7** [comment]
this bug is soved with CEF 71 Current version in master????? 

CEF 71 Current version in master #1235

**Comment by Julusian** [comment]
@Jgaleas7 No, this requires CEF74 but that version introduces other problems so isn't an option for now

**Comment by Jgaleas7** [comment]
Yhh

**Comment by Sidonai-1** [comment]
Any news on this issue?
